pkg/logs/input/file/tailer.go

```diff
@@ -124,14 +124,20 @@ func (t *Tailer) Start(offset int64, whence int) error {
 func (t *Tailer) readForever() {
 	defer t.onStop()
 	for {
+		n, err := t.read()
+		if err != nil {
+			return
+		}
+
 		select {
 		case <-t.stop:
+			if n != 0 && t.didFileRotate == 1 {
+				log.Warn("Tailer stopped after rotation close timeout with data remaining")
```

Comment:
Any suggestion for better wording here?

Reply:
```suggestion
				log.Warn("Tailer stopped after rotation close timeout with remaining unread data")
```

Reply:
Perhaps? Not really sure it's any better.